woof-code/rootfs-petbuilds/fixmenusd/fixmenusd.c

```diff
@@ -33,8 +37,40 @@ sh(const char *cmd, const sigset_t *set)
 		return -1;
 }
 
+static void
+setup_spot(const regex_t *re, const int usrbin, const struct inotify_event *event, const sigset_t *set)
+{
+	static char orig[NAME_MAX + sizeof(".bin")];
+	struct stat stbuf;
+	size_t len;
+	char *cmd;
+
+	if (regexec(re, event->name, 0, NULL, 0) != 0)
+		return;
+
+	if (fstatat(usrbin, event->name, &stbuf, AT_SYMLINK_NOFOLLOW) < 0)
+		return;
+
+	if (((event->mask & IN_CREATE) && !S_ISLNK(stbuf.st_mode)) ||
```

Comment:
Even links are handled correctly - for example, `ln -s /usr/bin/lxtask /usr/bin/microsoft-edge-stable` triggers `setup_spot microsoft-edge-stable=true` and it works